xpanse/api/attack_surface_rules/v1/attack_surface_rules.py

```diff
@@ -41,8 +42,23 @@ def list(
             >>> attack_surface_rules =  client.attack_surface_rules.list().dump()
         """
         kwargs = build_request_payload(request_data=request_data, **kwargs)
-        response = self._api.post(path=self.ENDPOINT, **kwargs)
-        return XpanseResponse(response, data_key=self.DATA_KEY)
+
+        # Use user provided `search_from` and `search_to` if provided in `request_data`, otherwise use default
```

Comment:
NIT: I'm no expert on Python syntax, but it just kinda feels weird to be manipulating a kwargs object don't it? Why not just pass those in as args directly into the XpanseResltIterator?

return XpanseResultIterator(
api=self.api,
path=self.ENDPOINT,
data_key=self.DATA_KEY,
use_page_token=False,
search_from=(request_data or {}).get(PublicApiFields.SEARCH_FROM, {}),
search_to=(request_data or {}).get(PublicApiFields.SEARCH_TO, {})
**kwargs,
)

or am I missing something here?

Reply:
I think it would override the defaults, but you make a good point I can just import the default const instead as the alternative